Src/Couchbase.Linq.IntegrationTests/BucketContextTests.cs

```diff
@@ -367,6 +367,26 @@ public void SubmitChanges_WhenDocsAreModifiedAndEndChangeTracking_DocumentIsNotM
             var doc = ClusterHelper.GetBucket("beer-sample").GetDocument<Beer>(((ITrackedDocumentNode)beer).Metadata.Id);
             Assert.AreNotEqual(beer.Abv, doc.Content.Abv);
         }
+
+        [Test]
+        public void Underlying_Bucket_can_be_used_to_execute_N1QL_query_directly()
+        {
+            // setup bucket context
+            var db = new BucketContext(ClusterHelper.GetBucket("beer-sample"));
+
+            // get a beer from the db
+            var beer = db.Query<Beer>().Select(b => N1QlFunctions.Meta(b)).First();
+
+            // execute N1QL to update the beer
+            var newBeerName = Guid.NewGuid().ToString();
+            var n1ql = string.Format("UPDATE `beer-sample` USE KEYS '{0}' SET name = '{1}';", beer.Id, newBeerName);
+            db.Bucket.Query<int>(n1ql);
```

Comment:
In this case Save(...) which uses K/V memcached protocol is faster/more efficient than going through N1QL over HTTP + all of the steps in-between before finally updating the document (using the same memcached protocol n the server). N1QL's only advantage here is SQL like syntax - but the path to get here is more complex. In fact, the motivation behind the Linq2CB was to abstract the N1QL syntax (which you cannot do 100% hence the N1QL helper functions) and make it easier for Linq2SQL and EF users to transition over to CB.